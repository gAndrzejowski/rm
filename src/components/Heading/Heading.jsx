import React from 'react';
import MovieHeader from '../MovieHeader/MovieHeader.jsx';
import SearchHeader from "../SearchHeader/SearchHeader";
import {Route} from 'react-router'

const Heading = () =>(
    <header>
        <Route exact path="/film/:id" component={MovieHeader} />
        <Route path="*" component={SearchHeader} />
    </header>
);
export default Heading;
