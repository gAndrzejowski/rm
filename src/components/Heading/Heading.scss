header {
  color:beige;
  padding:0 5vw 2em;
  max-width:1200px;
}
.top {
  display:flex;
  flex-flow:row nowrap;
  justify-content: space-between;
  align-items: center;
}