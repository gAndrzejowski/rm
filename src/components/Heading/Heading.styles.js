const styles = {
  '@global': {
    header: {
      color: 'beige',
      padding: [0, '5vw', '2em'],
      maxWidth: 1200,
    },
  },
  top: {
    display: 'flex',
    flexFlow: ['row', 'nowrap'],
    justifyContent: 'space-between',
    alignItems: 'center',
    '@global button': {
      color: 'firebrick',
      textTransform: 'uppercase',
      textAlign: 'center',
      display: 'block',
      backgroundColor: 'white',
      marginLeft: 10,
      border: 'none',
      padding: ['.2em', '1ch'],
      userSelect: 'none',
      outline: 'none',
      cursor: 'pointer',
      fontWeight: 'bold',
    },
  },
};
export default styles;
