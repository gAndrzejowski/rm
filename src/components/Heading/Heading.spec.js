import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Heading from "./Heading";

describe('Heading component', () => {
    const renderer = new ShallowRenderer();
    it('renders correctly when there is no movie chosen', () => {
        renderer.render(<Heading />);
        const tree = renderer.getRenderOutput();
        expect(tree).toMatchSnapshot();
    });
});

