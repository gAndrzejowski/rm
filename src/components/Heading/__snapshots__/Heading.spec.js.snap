// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading component renders correctly when there is no movie chosen 1`] = `
<header>
  <Route
    component={[Function]}
    exact={true}
    path="/film/:id"
  />
  <Route
    component={[Function]}
    path="*"
  />
</header>
`;
